File: Python/Day19_Feb10_2021/twoSum.py
    #Brute Force Method
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range(len(nums)):
            for j in range(len(nums)):
                if i != j:
                    if nums[i]+nums[j]==target:
                        return [i,j]
                    
        return []
    
    #Using hasmap
    def twoSum(self, nums, target):
        h = {}
        for i,val in enumerate(nums):
            diff = target - val
            if diff in h:
                return [h[diff],i]
            h[val] = i

File: C++/Day21_Feb14_2021/twoSum.cpp

class Solution {
public:
       //Using Brute Force
//     vector<int> twoSum(vector<int>& nums, int target) {
//         vector<int> out;
//         out.clear();
//         for(int i =0;i<nums.size();i++){
//             // cout<<nums[i]<<'\n';
//             for(int j = i+1;j<nums.size();j++){
//                 if(nums[i]+nums[j] == target){
//                     out.push_back(i);
//                     out.push_back(j);
//                     return out;
//                 }
//             }
//         }
        
//         return out;
        
//     }
    //Using hasmap : One pass
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> h;
        for(int i = 0;i<nums.size();i++){
            auto item = h.find(target - nums[i]);
            if( item != h.end()){
                return{item->second,i};
            }
            h[nums[i]] = i;
        }
        return{};
    }
};

